### PR TITLE
DAOS-8293 dfs: dfs_sys should not set size = -1

### PR DESCRIPTION
**Pull request by daltonbohning**
Test-tag: pr fs_copy dcp

Don't set size = -1 for daos_size_t types, since it is unsigned.
The rc is set appropriately anyway, so this is redundant.

Signed-off-by: Dalton Bohning <daltonx.bohning@intel.com>

### CODE REVIEW

File: src/include/daos_fs_sys.h [hunk 1]
```diff
@@ -27,15 +25,15 @@ extern "C" {
 #include <daos.h>
 #include <daos_fs.h>
 
-#define DFS_SYS_NO_CACHE 1
-#define DFS_SYS_NO_LOCK 2
+/** Mount flags for dfs_sys_mount. By default, mount with caching and locking turned on. */
+#define DFS_SYS_NO_CACHE 1 /** Turn off directory caching */
+#define DFS_SYS_NO_LOCK 2  /** Turn off locking. Useful for single-threaded applications. */
```
Comment (ashleypittman) [inline src/include/daos_fs_sys.h:30]: The annotation here is wrong, I think you need this
```suggestion
#define DFS_SYS_NO_LOCK 2  /**<Turn off locking. Useful for single-threaded applications. */
```

File: src/include/daos_fs_sys.h [hunk 2]
```diff
@@ -27,15 +25,15 @@ extern "C" {
 #include <daos.h>
 #include <daos_fs.h>
 
-#define DFS_SYS_NO_CACHE 1
-#define DFS_SYS_NO_LOCK 2
+/** Mount flags for dfs_sys_mount. By default, mount with caching and locking turned on. */
```
Comment (ashleypittman) [inline src/include/daos_fs_sys.h:28]: This comment is being picked up by doxygen, and applied only to DFS_SYS_NO_CACHE, so I think you want to remove the second * here and add a < on the line below.

File: src/include/daos_fs_sys.h (hunk 1 above)
Comment (daltonbohning) [reply src/include/daos_fs_sys.h]: Ah, Thanks!

File: src/include/daos_fs_sys.h (hunk 2 above)
Comment (daltonbohning) [reply src/include/daos_fs_sys.h:28]: Ah, Thanks!

Comment (daltonbohning) [reply src/include/daos_fs_sys.h:28]: Looking through the code I see other instances of `/**<` so that's that I went with

Comment (ashleypittman) [reply src/include/daos_fs_sys.h:28]: So I think now you've got three doxygen comments, one on L28 that doesn't apply to any code, then two on L29 and L30 that apply to the defines on those lines.  It's certainly enough to make doxygen happy but like all these things it'll be worth reading the generated html at some point.

Comment (daltonbohning) [reply src/include/daos_fs_sys.h:28]: Where can I find the generated html?

Comment (ashleypittman) [reply src/include/daos_fs_sys.h:28]: I don't think we save them in CI anywhere, but run "doxygen Doxyfile" and you should get a html directory to poke around in.

Comment (daltonbohning) [reply src/include/daos_fs_sys.h:28]: It seems the comment above and beside DFS_SYS_NO_CACHE are concatenated into a single doc comment.
image

If it's okay with you, I'd like to NOT push another commit just for this.

**Pull request merged**